test/test_maskedtensor.py

```diff
@@ -238,8 +237,276 @@ def test_masks_match(self, fn_name):
                 == str(e)
             )
 
+class TestReductions(TestCase):
+    def test_not_implemented(self):
+        d = torch.tensor([[0, 1, 2], [3, 4, 5.0]])
+        m = torch.tensor([[True, False, False], [False, True, False]])
+        mt = MaskedTensor(d, m)
+        self.assertRaises(TypeError, lambda: mt.max())
+
+    def test_sum(self):
+        d = torch.tensor([[0, 1, 2, 6], [3, 4, 5.0, 7]])
+        m = torch.tensor([[True, False, False, True], [False, True, False, True]])
+        mt = MaskedTensor(d, m)
+        _compare_mts(MaskedTensor(torch.tensor(17.0), torch.tensor(True)), mt.sum())
+        _compare_mts(
+            MaskedTensor(
+                torch.tensor([0.0, 4.0, 1.0, 13]),
+                torch.tensor([True, True, False, True]),
+            ),
+            mt.sum(dim=0),
+        )
+
+    def test_sum_grad(self):
+        d = torch.tensor([[0, 1, 2], [3, 4, 5.0]])
+        m = torch.tensor([[True, False, False], [False, True, False]])
+        mt = MaskedTensor(d, m, requires_grad=True)
+        mt.sum().backward()
+        _compare_mts(mt.grad, MaskedTensor(torch.tensor(1.0).expand_as(m), m))
+
+    def test_mean(self):
+        d = torch.tensor([[0, 1, 3, 2], [3, 4, 1.0, 4]])
+        m = torch.tensor([[True, False, False, True], [False, True, False, True]])
+        mt = MaskedTensor(d, m)
+        _compare_mts(MaskedTensor(torch.tensor(2.5), torch.tensor(True)), mt.mean())
+        _compare_mts(
+            MaskedTensor(
+                torch.tensor([0.0, 4.0, 1.0, 3]),
+                torch.tensor([True, True, False, True]),
+            ),
+            mt.mean(dim=0),
+        )
+
+    def test_mean_grad_case_1a(self):
```

Comment:
A comment above this block of tests will unlock a lot of the value of your recent voyage of discovery 😁  Specifically, it should:
- clarify that you're using `mean()` as a proxy to test the different combinations of MT constructors and `requires_grad` settings of the MT and its `data` argument (i.e.,. it's not behavior that's specific to `mean()`)
- recap the expected use of the `MaskedTensor()` constructor and `from_values()`, wrt whether the resulting MT is a leaf node (i.e. whether or not gradient should flow through to the `data` argument) 

Doesn't have to be long, but putting this information next to the tests lets a reader know why they're the way they are, without doing a bunch of digging.

Reply:
thanks for the comment, added a long comment :)